src/irsForms/ScheduleEIC.ts

```diff
@@ -1,11 +1,355 @@
-import { TaxPayer } from '../redux/data'
+import { Dependent, FilingStatus, TaxPayer as TP } from '../redux/data'
+import TaxPayer from '../redux/TaxPayer'
+import F1040 from './F1040'
+import { sumFields } from './util'
+import * as federal from '../data/federal'
+import F2555 from './F2555'
+import F4797 from './F4797'
+import F8814 from './F8814'
+import Pub596Worksheet1 from './worksheets/Pub596Worksheet1'
+import Form from './Form'
+import { anArrayOf, evaluatePiecewise, Piecewise } from '../util'
 
-export default class ScheduleEIC {
+type PrecludesEIC<F> = (f: F) => boolean
+
+const unimplemented = (message: string): void =>
+  console.warn(`[Schedule EIC] unimplemented ${message}`)
+
+const checks2555: PrecludesEIC<F2555> = (f): boolean => {
+  unimplemented('check F2555')
+  return false
+}
+
+const checks4797: PrecludesEIC<F4797> = (f): boolean => {
+  unimplemented('check F4797')
+  return false
+}
+
+const checks8814: PrecludesEIC<F8814> = (f): boolean => {
+  unimplemented('check F8814')
+  return false
+}
+
+const checksPub596: PrecludesEIC<Pub596Worksheet1> = (f): boolean => {
+  unimplemented('check Pub596 worksheet 1')
+  return false
+}
+
+const precludesEIC = <F>(p: PrecludesEIC<F>) => (f: F | undefined) => {
+  if (f === undefined) {
+    return false
+  }
+  return p(f)
+}
+
+export default class ScheduleEIC implements Form {
   tp: TaxPayer
+  f2555?: F2555
+  f4797?: F4797
+  f8814?: F8814
+  pub596Worksheet1: Pub596Worksheet1
+  qualifyingStudentCutoffYear: number = 1996
+  qualifyingCutoffYear: number = 2001
+  investmentIncomeLimit: number = 3650
+
+  constructor (tp: TP) {
+    this.tp = new TaxPayer(tp)
+    this.f2555 = new F2555(tp)
+    this.f4797 = new F4797(tp)
+    this.f8814 = new F8814(tp)
+    this.pub596Worksheet1 = new Pub596Worksheet1(tp)
+  }
+
+  // At least key children, and max [NonMFJLimit, MFJLimit] income
+  step1ChildrenIncomes: Array<[number, number]> = [
```

Comment:
Code nit: I can't find where this variable is used. It looks similar to the line11caps in federal.ts. Is it needed?

Reply:
Nice catch, maybe the linter doesn't pick up unused class members like this.